components/mailchimp/mailchimp.app.js

```diff
@@ -0,0 +1,681 @@
+     * Reference parameters of sub-objects with dot notation.
+     * @returns {stores: array, total_file_size: integer, total_items: integer, _links: object } An
+     * array with the information of the `stores` returned, `total_items` with the total count of
+     * the stores collection, and `_links` array of link types and descriptions for the API schema
+     * documents.
+     */
+    async getAllStores(config) {
+      return await this._withRetries(() =>
+      this.api().ecommerce.stores(config));
+    },
+    /**
+     * Gets orders in the specified Mailchimp connected acccount's Ecommerce Store.
+     * @param {String} storeId - The unique ID of the Ecommerce Store you'd like to get
+     * orders from.
+     * @param {Object} config - An object representing the configuration options for this method.
+     * @param {Integer} config.count - For pagination, the number of records to return on each page.
+     * Default value is 10. Maximum value is 1000.
+     * @param {Integer} config.offset - For pagination, this the number of records from a collection
+     * to skip. Default value is 0.
+     * @param {String} config.campaignId - Restrict results to orders with this specific campaignId
+     * value.
+     * @param {String} config.outreachId - Restrict results to orders with a specific outreachId
+     * value.
+     * @param {String} config.customerId - Restrict results to orders made by the specific customer
+     * with this unique ID.
+     * @param {Boolean} config.hasOutreach - Restrict results to orders that have an outreach
+     * attached.
+     * For example, an email campaign or Facebook ad.
+     * @returns {store_id: string, orders: array, total_items: integer, _links: object } `store_id`
+     * with the unique ID of the Ecommerce Store if specified (not present otherwise), an array
+     * with the information of the `orders` returned, `total_items` with the total count of the
+     * orders collection, and `_links` array of links for order manipulation through the Mailchimp
+     * API.
+     */
+    async *getOrderStream(storeId, config) {
+      let offset = config.offset;
+      config.campaignId = config.campaignId === ""
+        ? null
+        : config.campaignId;
+      config.outreachId = config.outreachId === ""
+        ? null
+        : config.outreachId;
+      config.customerId = config.customerId === ""
+        ? null
+        : config.customerId;
+      config.hasOutreach = config.hasOutreach === "Both"
+        ? null
+        : config.hasOutreach;
+        do {
+          const opts = {
+            ...config,
+            offset,
+          };
+          const { orders = [] } = await this._withRetries(
+            () => this.api().ecommerce.getStoreOrders(storeId, opts),
+          );
+          if (orders.length === 0) {
+            return;
+          }
+          for (const order of orders) {
+            yield order;
+          }
+        offset += orders.length;
```

Comment:
The indentation looks a bit off here.
```suggestion
      config.hasOutreach = config.hasOutreach === "Both"
        ? null
        : config.hasOutreach;
      do {
        const opts = {
          ...config,
          offset,
        };
        const { orders = [] } = await this._withRetries(
          () => this.api().ecommerce.getStoreOrders(storeId, opts),
        );
        if (orders.length === 0) {
          return;
        }
        for (const order of orders) {
          yield order;
        }
        offset += orders.length;
```